System/OsPath/Data/ByteString/Short/Word16.hs

```diff
@@ -647,6 +650,23 @@ splitWith p = \(assertEven -> sbs) -> if
               | otherwise -> a : go (tail b)
 
 
+-- | Check whether one string is a substring of another.
+isInfixOf :: ShortByteString -> ShortByteString -> Bool
+isInfixOf sbs = \s -> null sbs || not (null $ snd $ GHC.Exts.inline breakSubstring sbs s)
+
+
+breakSubstring :: ShortByteString -- ^ String to search for
+               -> ShortByteString -- ^ String to search in
+               -> (ShortByteString, ShortByteString) -- ^ Head and tail of string broken at substring
+breakSubstring pat = go mempty
+ where
+    go prefix s = let (h, t)  = GHC.Exts.inline BS.breakSubstring pat s
+                      prefix' = prefix `append` h
+                  in if | even (BS.length prefix') -> (prefix', t)
+                        | otherwise -> let (p, s') = BS.splitAt (BS.length h + 1) s
+                                       in go p s'
```

Comment:
I'm just eyeballing this code, so I may miss the obvious here, but shouldn't this be `go (prefix <> p) s'`?